### ISSUE TITLE
sensitive information exposed in `debug` level logs

### ISSUE THREAD

**Issue by KevinWang15** [saltstack/salt#58298]
**Description**

When debug level logging is enabled, sensitive information / credentials are exposed and leaked in the logs:

```
[salt.loaded.int.render.yaml:68  ][DEBUG   ][11790] Results of YAML rendering: ...
[salt.template    :113 ][DEBUG   ][11790] Rendered data from file: ...
```

**Expected behavior**

I believe that the values should be redacted **by default**. Some users do not check the log files and do not know that the credentials are being leaked. Only displaying the `first 3 characters` of any value, followed by `** hidden **` should be good enough for logging and debugging purposes (an environment variable could be provided if the user wants to explicitly disable the redaction).

**Comment by dwoz** [comment]
I agree, actually I don't think we should even be redacting credentials in logs. A more secure practice is to *never* log credentials.

**Comment by dwoz** [comment]
The two log lines cited here are handled on all supported branches (3006.x / 3007.x / 3008.x / master):

- `salt.loaded.int.render.yaml` DEBUG `Results of YAML rendering: ...` — removed in b64c6343e26 ("Removing unneccessary loggint that could display sensitive data").
- `salt.template` DEBUG `Rendered data from file: ...` — gated behind `logging.GARBAGE` (level 1), not DEBUG, and additionally passed through `salt.utils.sanitizers.mask_args_value` before emission.

Closing as already fixed. Please reopen with a fresh reproduction on a current branch if you still see sensitive values in DEBUG logs.
